### PR TITLE
Gelato scheduler automation

### PR DESCRIPTION
**Pull request by joeljesu**
Adding in a simple Gelato resolver contract to make it easier to automate Flow Scheduling with Gelato.

Resolver contract contains a `checker` function that returns the appropriate transaction payload (`executeCreateFlow`, `executeDeleteFlow`, or nothing) so Gelato can execute appropriately.

### CODE REVIEW

**Comment by joeljesu** [comment]
Looking at information from @donoso_eth's [Gelato v2 x Superfluid guide](https://medium.com/coinmonks/gelato-v2-superfluid-eb13166ac414), this can be improved further before merging.

File: packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol
Comment (kasparkallas) [inline packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol]: I feel like the condition for "create flow" should be before "delete flow".

File: packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol
Comment (kasparkallas) [inline packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol]: "FlowSchedulerGelatoResolver" or something more specific could be better?

File: packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol
Comment (joeljesu) [reply packages/automation-contracts/scheduler/contracts/SchedulerGelatoResolver.sol]: I was under the impression there could be a situation where flowSchedule.startDate + startMaxDelay is greater than flowSchedule.endDate and in that case, the resolver could return a start payload when it shoud be a delete payload if the condition for start came before delete

However, looking here: https://github.com/superfluid-finance/protocol-monorepo/blob/gelato-scheduler-automation/packages/automation-contracts/scheduler/contracts/FlowScheduler.sol#L158 - if the flow was started, then flowSchedule.startDate would be zeroed out and returning a start payload would be blocked by `flowSchedule.startDate != 0` on line 52. 

I can make this adjustment!

**Comment by ngmachado** [comment]
Already talked with @sunnyjaycer about this PR. 

## Recap:
- Check GelatoV2
- Rename canonical name to real name flowScheduler.
- Review tests and remove tests not related to gelato integration. _(if needed to add more tests do it in the proper test file)_
- Take into consideration @kasparkallas naming proposal.

**Comment by joeljesu** [comment]
Keeping Gelato v2 out of scope here, will do some homework and incorporate support later

**Comment by ngmachado** [comment]
> Looking at information from @donoso_eth's [Gelato v2 x Superfluid guide](https://medium.com/coinmonks/gelato-v2-superfluid-eb13166ac414), this can be improved further before merging.

It was you that brought this to scope 😬

going to finish my review tomorrow. Thank you for the fast changes!

File: packages/automation-contracts/scheduler/contracts/FlowSchedulerResolver.sol
Comment (ngmachado) [inline packages/automation-contracts/scheduler/contracts/FlowSchedulerResolver.sol]: remove

File: packages/automation-contracts/scheduler/old-resolver-test.txt
Comment (ngmachado) [inline packages/automation-contracts/scheduler/old-resolver-test.txt]: is this file needed? if not delete it

File: packages/automation-contracts/scheduler/test/FlowSchedulerResolver.t.sol
Comment (ngmachado) [inline packages/automation-contracts/scheduler/test/FlowSchedulerResolver.t.sol]: delete it

**Review by ngmachado — CHANGES_REQUESTED** [review]

**Review by ngmachado — APPROVED** [review]

**Comment by hellwolf** [comment]
@ngmachado any holdup for merging?

**Comment by ngmachado** [comment]
@hellwolf for me is good to merge. We need another approve to merge it.

**Comment by hellwolf** [comment]
image

@ngmachado you'd need to squash and force-push

**Pull request merged**
